presto-hive-metastore/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java

```diff
@@ -40,23 +41,38 @@
     private final List<SortingColumn> sortedBy;
     private final BucketFunctionType bucketFunctionType;
     private final Optional<List<Type>> types;
+    private final Optional<List<String>> clusteredBy;
+    private final Optional<List<Integer>> clusterCount;
+    private final Optional<List<String>> distribution;
 
     @JsonCreator
     public HiveBucketProperty(
             @JsonProperty("bucketedBy") List<String> bucketedBy,
             @JsonProperty("bucketCount") int bucketCount,
             @JsonProperty("sortedBy") List<SortingColumn> sortedBy,
             @JsonProperty("bucketFunctionType") BucketFunctionType bucketFunctionType,
-            @JsonProperty("types") Optional<List<Type>> types)
+            @JsonProperty("types") Optional<List<Type>> types,
+            @JsonProperty("clusteredBy") Optional<List<String>> clusteredBy,
```

Comment:
Since we already have `BucketFunctionType`, do we still need to differentiate `bucketedBy` <-> `clusteredBy` and `bucketCount` <-> `clusterCount`? Logically they are doing the same thing, which is figuring out which columns should we bucket on and by how many buckets, and apply the bucket function as specified.